### PR TITLE
Iron deficiency anemia update

### PR DESCRIPTION
**Pull request by alibow**
This PR reflects the updated definition of the iron deficiency risk factor from my previous PR, adds modeling strategy and some limitation. V&V to come.

### CODE REVIEW

**Comment by yaqi7** [comment]
@alibow For the description section, it would be better if citations could be added.

**Comment by NathanielBlairStahn** [comment]
Hi Ali, I just noticed that in your cause hierarchy diagram, the cause is listed as "iron deficiency" instead of "dietary iron deficiency"

**Comment by NathanielBlairStahn** [comment]
We should write down the equations for the gamma and mirror Gumbel distributions, so we can see what all the parameters mean and verify that the python code is calculating the right things. And like Beatrix just suggested, we should make sure there's a link to the R code actually used by the anemia team.

**Comment by NathanielBlairStahn** [comment]
@alibow I just noticed that the document's internal hyperlink target `.. _2017 cause iron deficiency:` is still missing underscores.

File: docs/source/gbd2017_models/risk_attributable_causes/iron_deficiency/index.rst [hunk 1]
```diff
@@ -307,25 +318,84 @@ Risk Factor Hierarchy
...
+	:widths: 5, 30, 20
+	:header-rows: 1
 
-State and Transition Data Tables
-++++++++++++++++++++++++++++++++
+	* - Anemia Severity
+	  - All Anemia Sequela
+	  - Iron Responsive Anemia Sequela
+	* - Mild
+	  - 144, 172, 177, 182, 206, 240, 438, 442, 525, 531, 537, 645, 648, 651, 654, 1004, 1008, 1012, 1016, 1020, 1024, 1028, 1032, 1057, 1061, 1065, 1069, 1079, 1089, 1099, 1106, 1120, 1361, 1373, 1385, 1397, 1409, 1421, 1433, 1445, 4952, 4955, 4976, 4985, 4988, 5009, 5018, 5027, 5036, 5051, 5063, 5075, 5087, 5099, 5111, 5123, 5225, 5228, 5249, 5252, 5273, 5276, 5393, 5567, 5579, 5606, 5627, 5648, 5651, 5654, 5678, 5699, 5702, 5705, 7202, 7214, 22989, 22990, 22991, 22992, 22993, 23030, 23034, 23038, 23042, 23046
```
Comment (SantoniS) [inline docs/source/gbd2017_models/risk_attributable_causes/iron_deficiency/index.rst]: I prefer this format of your severity splits and will adopt this format for the causes I'm documenting with severity splits.

File: docs/source/gbd2017_models/risk_attributable_causes/iron_deficiency/index.rst [hunk 2]
```diff
@@ -307,25 +318,84 @@ Risk Factor Hierarchy
 
 .. image:: iron_risk_hierarchy.svg
 
-Cause Model Diagram
--------------------
 
-Data Description
-----------------
+Modeling Strategy
```
Comment (SantoniS) [inline docs/source/gbd2017_models/risk_attributable_causes/iron_deficiency/index.rst]: Is this 'Vivarium Modeling Strategy'?

Comment (alibow) [reply docs/source/gbd2017_models/risk_attributable_causes/iron_deficiency/index.rst]: Yes! I updated the headings/structure to make this more clear. Thanks :)

File: docs/source/gbd2017_models/risk_attributable_causes/iron_deficiency/index.rst [hunk 3]
```diff
@@ -151,9 +153,172 @@ described in the table above.
...
+Gumbel ensemble distribution. 
+
+NOTE:
+
+	The values stored in the iron deficiency risk factor (i.e. population  
+	hemoglobin concentration parameters) are used in the GBD modeling process 
+	to calculate risk-deleted population hemoglobin concentration where the 
+	risk is all iron deficiency (i.e. iron responsive anemias), which serves as 
+	the population TMREL, in order to calculate the population attributable 
+	fraction between the iron deficiency risk factor and maternal disorder
```
Comment (NathanielBlairStahn) [inline docs/source/gbd2017_models/risk_attributable_causes/iron_deficiency/index.rst]: This sentence is a bit hard to follow because it's so long.  Perhaps split it into a couple sentences?

File: docs/source/gbd2017_models/risk_attributable_causes/iron_deficiency/index.rst [hunk 4]
```diff
@@ -151,9 +153,172 @@ described in the table above.
...
+	The values stored in the iron deficiency risk factor (i.e. population  
+	hemoglobin concentration parameters) are used in the GBD modeling process 
+	to calculate risk-deleted population hemoglobin concentration where the 
+	risk is all iron deficiency (i.e. iron responsive anemias), which serves as 
+	the population TMREL, in order to calculate the population attributable 
+	fraction between the iron deficiency risk factor and maternal disorder 
+	causes. See the GBD 2017 Risk Factor Methods Appendix for more information. 
+	However, this process is not relevant for the simulation science team use 
+	of the iron deficiency risk factor as it relates to the dietary iron 
+	deficiency cause in GBD 2017.
```
Comment (NathanielBlairStahn) [inline docs/source/gbd2017_models/risk_attributable_causes/iron_deficiency/index.rst]: This last sentence is a bit unclear.  Do you mean that the above process is irrelevant because we're not modeling maternal disorders, only dietary iron deficiency (which is PAF-of-1, so there's no need to calculate a TMREL)?

In which case, the process would become relevant to our team if we ever wanted to model maternal disorders and hemoglobin levels in the same simulation.

Comment (alibow) [reply docs/source/gbd2017_models/risk_attributable_causes/iron_deficiency/index.rst]: Yes, that is what I mean. I just wanted to note that this happens, but it's not relevant to/within the scope of this PAF-of-1 cause model document, specifically.

Comment (alibow) [reply docs/source/gbd2017_models/risk_attributable_causes/iron_deficiency/index.rst]: I took out the "simulation science team" part and clarified this paragraph a bit.

**Comment by NathanielBlairStahn** [comment]
Hey @alibow , this looks good.  Thanks for all your work!  We just need one more approval and we can merge it into master.

File: docs/source/gbd2017_models/risk_attributable_causes/iron_deficiency/index.rst [hunk 5]
```diff
@@ -55,7 +54,8 @@ to all causes modeled in GBD (ex: dietary iron deficiency anemia, anemia due to
 maternal hemorrhage, sickle cell anemia, etc.). Estimating the total prevalence 
 of the anemia impairment for a given population is the first step in modeling 
 anemia in GBD 2017. This is done by fitting a distribution of hemoglobin levels 
-for that population from primary input data. For GBD 2017, an ensemble 
+for that population from primary input data based on the population's 
+hemoclobin concentration mean and standard deviation. For GBD 2017, an ensemble
```
Comment (aflaxman) [inline docs/source/gbd2017_models/risk_attributable_causes/iron_deficiency/index.rst]: typo, hemo*g*lobin

File: docs/source/gbd2017_models/risk_attributable_causes/iron_deficiency/index.rst [hunk 6]
```diff
@@ -208,27 +256,79 @@ conditions that cause iron deficiency anemia.* Below is a list of such causes, a
...
+
+Hemoglobin Concentration
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+In order to initialize an individual's hemoglobin concentration, each simulant 
+should be assigned a random number between 0 and 1 (random_number_i). This 
+number will represent the percentile of hemoglobin concentration for that 
+individual simulant relative to the baseline population distribution of 
+hemoglobin concentrations (from the GBD iron deficiency risk factor rei_92) *
+for the remainder of the simulation*. The corresponding hemoglobin concetration
```
Comment (aflaxman) [inline docs/source/gbd2017_models/risk_attributable_causes/iron_deficiency/index.rst]: typo conce*n*tration

File: docs/source/gbd2017_models/risk_attributable_causes/iron_deficiency/index.rst [hunk 7]
```diff
@@ -208,27 +256,79 @@ conditions that cause iron deficiency anemia.* Below is a list of such causes, a
...
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+In order to initialize an individual's hemoglobin concentration, each simulant 
+should be assigned a random number between 0 and 1 (random_number_i). This 
+number will represent the percentile of hemoglobin concentration for that 
+individual simulant relative to the baseline population distribution of 
+hemoglobin concentrations (from the GBD iron deficiency risk factor rei_92) *
+for the remainder of the simulation*. The corresponding hemoglobin concetration 
+for that percentile should then be assigned to the simulant using the 
+methodology described in the reaminder of this section.
```
Comment (aflaxman) [inline docs/source/gbd2017_models/risk_attributable_causes/iron_deficiency/index.rst]: typo re*ma*inder
